src/helpers/parser.ts

```diff
@@ -30,7 +30,7 @@ const formatKeyEntry = ({
   let line = "";
 
   if (key) {
-    line += key;
+    line += JSON.stringify(key);
```

Comment:
I dont think we should be using `JSON.stringify` here. `key` is simply a string, by stringifying as if its JSON we could add unintended characters. Why not just wrap the string: `"${key}"`

Also, rather than always wrapping the key in quotations, how about we only do it if the `key` includes `-`? Is there any other cases where we would need this?

Reply:
keys might get funky, not only with "-"

https://jira.mongodb.org/browse/SERVER-3229?focusedCommentId=36821&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-36821

According to that reply, we can have any UTF8 character, so... making that check would be quite the regex. We can always wrap the key anyway.

Wrapping the key string with your idea `${key}` won't work: it's the same as `key`. We need to transform it by always inserting `"` or `'` around it. The problem then comes with the special characters, for example, if we wrap it with `"` and the key contains `"` characters, or we wrap it with `'` and the key contains `'` characters. 
Hence the `JSON.stringify` solution: it will always wrap it around `"` and will always escape the `"` characters that might be in the key.

PS: updated the PR without the `package-lock.json` change

Reply:
Interesting, yep you're right. For the `${key}` solution, I meant to do something like this

```
key = `"${key}"`
```

just looked weird with MD formatting. But I didn't realize MongoDB supported so many special characters.

Is there any other possible side effects with using `JSON.stringify`? Still feel a bit dangerous to do, especially indiscriminately on every field. But if you have a source that proves there are no other harmful side effects then I'm good to go ahead with it.

In terms of characters that need to be escaped, I believe its only `"` (or if we want to wrap keys with single quotes, then its only `'`). If that's the case, I think it may still be safer to handle this manually. What about something like:

```
key = key.replace(/'/, `\'`);
key = `'${key}'`
```

Reply:
![Selection_002](https://user-images.githubusercontent.com/31771519/164472837-ad8df993-c0c7-43c8-9c11-42d75b134bfb.png)

I thought about that as well.

But I'm not sure, as you can see by the end result, there is no `\'` in the place of the original `'`, so I don't think this would work.

What do you think?

Reply:
Yeah this is tricky... I found this thread which may actually work in this case. I've never heard of this `String.raw` syntax before, but it looks promising. https://stackoverflow.com/a/10042082/8292279 

Dont have much time to look into it now but I will on the weekend. I think thats our best bet for now so that we dont have to use JSON.stringify, let me know your thoughts.

Reply:
@NelsonFrancisco Have you had a chance to check out the method mentioned above?

Reply:
Sorry. No time so far

But here's a REPL
![Selection_002](https://user-images.githubusercontent.com/31771519/165314070-57cc351a-1294-4305-9781-342e18fde74c.png)


Not sure what's going on here. But outputs are similar

Not sure about it though `"hey\\'man\\'"`: looks like two `\\` are there two times to escape the `\` and make the final string `hey\'man\'` which looks ok I guess

If we wrap that around `'` with `` line = '${result}' `` it might work :shrug:

Reply:
Anyway, just made a test with the first solution. E2E.

Looking good. Place special characters in a key, including `'` and works just fine

Curious that the output file only enclosures keys with `'` if needed. Normal character keys do not have it :thinking:

Reply:
Hmm, Im still uncertain if `JSON.stringify` could have other unintended consequences? Is there any other characters that it might wrongly escape or encode? 

I think with `String.raw` we should be able to leave out any escaping `\` like so:
<img width="306" alt="Screen Shot 2022-04-28 at 10 22 29 AM" src="https://user-images.githubusercontent.com/26126685/165774323-22468e08-574c-4a8d-99ec-0518ff00d8a7.png">

Reply:
When I said "first solution" I meant `` key.replace(/'/g, `\\\'`) ``

Reply:
Seems like `JSON.stringify` is still being used above. Any reason why you don't like the `String.raw` approach?

Reply:
I didn't update the PR yet

The `String.raw` is as any good as the `key.replace`

It just seems more uncommon, but I'm fine with it as well

Reply:
@NelsonFrancisco Looking to still merge this in, can you replace the `JSON.stringify` solution with the `String.raw` solution?

Reply:
@francescov1 do we just need to replace this line:
```
line += JSON.stringify(key);
```
with this line?
```
line += String.raw`${key}`;
```
I would make the change but I don't think I can edit this PR and would have to create a new one.  Do you have the privileges to do it?  I'd love to use this.  😀

Reply:
I used `line += String.raw`${key}`;` on my code and it didn't work.  I then tried the `line += JSON.stringify(key);` solution and it did work.

Reply:
I agree it's not ideal to enclose every single key with quotations but it's also a generated file so I don't see it as an issue.

I think your regex fix looks good but I personally don't think it's a problem worth fixing.  I would leave the PR as is.

Reply:
Also, we might need to re-generate the `/helpers/tests/artifacts/*` files to pass tests.

Reply:
I ran a couple of tests and I have good news.  Using `line += JSON.stringify(key);` does not wrap all the other keys in quotes.  Here's a snippet of what it generates:

```
  message?: string;
  messages: {
    default?: string;
   ...
    zh?: string;
    "zh-TW"?: string;
    ko?: string;
  };
  title?: string;
  event?: string;
  days?: number;
```

Reply:
It should wrap them all in quotes so I guess something is removing those quotes later? https://jsfiddle.net/nvsgd1yx/1/

Reply:
I ran a test and found that _it did_ wrap each one. Maybe you have a linter that removes them? Anyhow, lets do this PR with the regex I mentioned above. @wootwoot1234 do you want to create a new PR with that change, or shall I?

Reply:
Do you mind doing it?  I think you should double-check the tests too.

Reply:
@wootwoot1234 https://github.com/francescov1/mongoose-tsgen/pull/100